packages/vite/src/node/config.ts

```diff
@@ -362,7 +364,10 @@ export async function resolveConfig(
     root: resolvedRoot,
     base: BASE_URL,
     resolve: resolveOptions,
-    publicDir: path.resolve(resolvedRoot, config.publicDir || 'public'),
+    publicDir:
+      config.publicDir !== false
```

Comment:
**thought:** On all places within this PR we are checking `config.publicDir` with existente-checks instead of `!== false` check.
We could do this also here, so if (for any reasons) `config.publicDir` is an empty string, it counts also as false

```suggestion
      !!config.publicDir
```

Reply:
@Shinigami92 
Do you mean if users set `publicDir` to an empty string then do not process static files? The default is `public` only when the user does not set `publicDir`.

Reply:
Should `config.publicDir` handled as follows?
 `false` - do not process static files 
` true`  - set default 'public'
` ''`    - do not process static files 
 not set value - set default 'public'

Reply:
IMO '' should not be counted as false here. It should be an error for the user, that used a wrong pathname.

Reply:
@patak-js Yeah, that could be one of two possible options

Either throw/log a warning/error or handle it consistently with all other `if`-branches

But the current implementation would result in weird results cause here only `false` is checked and then below suddenly also against falsy values is checked.

_And yes, I'm talking about user that set `publicDir: ''`, the type-definition `string` doesn't deny that_

Reply:
The current version of `publicDir:''` will be set to default value  `public`, do we need to change this behavior?

Reply:
I think that is a bug, maybe we can fix it in another PR for better tracking. I don't think nobody is using this undocumented behavior, but just in case.

Reply:
I will modify the current PR as follows，right？
`publicDir:false`   -- do not process static files
`publicDir:true`    -- set default 'public'
`publicDir:''`         -- set default 'public'
not set value        -- set default 'public'

Next, I will fix  `publicDir:''` and `publicDir:true` in another PR ，Throw a warning or error to remind the user to set the correct value.

Reply:
Sound good to me. `publicDir: true` is correct here though, it should use the default 'public'